R/install_reqs.R

```diff
@@ -0,0 +1,211 @@
+install_special_req = function(elem, pattern, f, verbose){
+  #' @param elem list element to be worked on
+  #' @param pattern regex pattern to match on
+  #' @param f the function used to install
+  #' @return count of failures
+  
+  failures = 0
+  
+  if(length(elem) > 0){
+    for(i in elem){
+      url = sub(pattern,'',i)
+      v = vmess(sprintf(INSTALL,url,''),verbose)
```

Comment:
`INSTALL` is not defined in this scope.  Same for all other constants that are used for messages (ie `OTHER_FAIL`, `INSTALL_VERSION`).

Reply:
Another 🤦‍♂️ from me.  They are defined in the parent env of the only function that should be calling these helpers (I think)?

I'm going to make these message constants globals.

There are lingering scope issues in the `devtools::check` that are not related to these constants:

```
❯ checking R code for possible problems ... NOTE
  ...
   Undefined global functions or variables:
     comp dryrun i package repo
```

These feel like they should be passed as arguments rather than pulled from the parent environment.

Reply:
Good catch - I need to do better about actually running R CHECK

Reply:
Would it make sense to add them to package data?

Reply:
I’m not too sure the dis/advantages of moving/not moving. 

If we do move them I think they should be renamed more explicitly since they won’t have the same context that helps understand them.